### DISCUSSION TITLE
Table Design

### DISCUSSION THREAD

**Discussion by PGLange** [Budibase/budibase#1821] [Help] [answered]
Hi again,
I'm testing out the possibilities that tables offer. 
- For Layout reasons I need to define the width for table cells, this becomes urgend when there are more than a few fields (in my tests there are 14 fields). 

- Is there a possibility to change the order of the collums on the screen?

- Is it possible to use filter-fields, like a search-field? Or a Dropdown-field to manipulate the table?

**Comment by mjashanks** [comment, marked as answer]
Hey @PGLange.

Unfortunately, its not currently possible to change the width or order of the columns.

As for adding a filter:

1. Surround all of your components in a `Form` component. In the following screenshot, I've named mine "Search Form"
2. Add a new Text field, and be sure to type a value for `field` - we will use this later for the filter. Unfortunately, it's not possible to use a dropdown for a filter, as our dropdowns do not currently allow us to enter custom options (they only work with options declared on a column in an internal table)

![image](https://user-images.githubusercontent.com/3524181/123241744-51c83880-d4d9-11eb-9632-8a66d10ce55d.png)

3. Click on your `Data Provider` and then `Define Filters`. In here you can apply a filter, using the field value that you created above.

![image](https://user-images.githubusercontent.com/3524181/123242105-9bb11e80-d4d9-11eb-8006-1da5028a9aa7.png)

In your app, you just have to type into the field for the filter to be applied.

Im converting this discussion to an Idea, as everything you have suggested in here is a sensible request!

**Reply by mjashanks** [reply]
@PGLange I stand corrected!

You can change the order of the columns by 
1. Unselecting them all
2. Selecting them in the order that you want them to appear.

**Reply by Cinbert** [reply]
Now I'm confused.  I am also struggling with a filtered browse screen. I was following the YouTube tutorial [How to: Dynamic filtering of data using forms and dataproviders] In that  tutorial you can choose the new field for the binding  from the list of columns. [here](https://youtu.be/hmmLdeY-5d4?t=230) That wasn't working for me as my new fields were not showing up in that list. Therefore I was happy to find this example here.  

But in the example above it appears that you can not choose the field (created earlier) but have to type it in to complete  the binding statement.  If you type it in, what is the syntax for that entry?  In your example it is {{Search Form.Fields.nameSearch}}  nameSearch was defined as the search field earlier, Search Form is the name of the form, so what is "Fields"?  Is that always the same?  If I follow that syntax my string is: {{ Search Form.Fields.typeSearch }}  which gives me an error "Current Handlebars syntax is invalid ..."

Is there possibly something I need to do to refresh things or somehow register those new field names? Is the video still valid?   I have updated to the latest Current Version: 0.9.56 and started again from scratch after the update.  Thank you for your help.

**Reply by mike12345567** [reply]
Hi @Cinbert one thing to think about here is the order of the components in the tree, if your form fields aren't showing up within your data provider binding options then it is likely the data provider is not within the form component.

Here you can see how it should look in your tree, the data provider within your form.
![image](https://user-images.githubusercontent.com/4407001/123264586-138b4300-d4f2-11eb-87c0-6d99d0b665e5.png)

**Reply by Cinbert** [reply]
@mike12345567 That was it.  I tried to see what was contained within what in the YouTube video but I missed that detail. Once I put the data provider inside the form component the fields showed up for selection in the binding. Thanks for noticing what wasn't addressed in the earlier reply.

**Comment by PGLange** [comment]
@mjashanks 
thanks for answering - that would have taken a long time if I'd had to play the possibilities out!
Great - Thank you :)

**Reply by mjashanks** [reply]
@Cinbert You're correct - the way that the video suggests also works for me - I've just tested that. I am using version `0.9.58`. Not sure if this helps, but I was initially tripped up when testing this:
- I clicked `Define Fillters` 
- then the ⚡ button to open the bindings
- Set my bindings
- Clicked `Save` to save the binding
I then forgot to press save again, to save my filter.

And...

>  If I follow that syntax my string is: {{ Search Form.Fields.typeSearch }} which gives me an error "Current Handlebars syntax is invalid ..."

Your field `typeSearch` should appear in your list of available bindings - make sure you use this rather than typing manually.

![image](https://user-images.githubusercontent.com/3524181/123264131-a1b2f980-d4f1-11eb-826c-90c399dd8b8d.png)
